.circleci/config.yml

```diff
@@ -112,16 +112,17 @@ references:
                   name: Build Workspace | << parameters.workspace >>
                   working_directory: << parameters.workspace >>
                   command: |
-                    if [ -d install ] && [ ! -f build_failed ]
+                    BUILD_UNFINISHED=$(colcon list --packages-skip-build-finished)
+                    BUILD_FAILED=$(colcon list --packages-select-build-failed)
+                    if [ -n "$BUILD_UNFINISHED" ] || [ -n "$BUILD_FAILED" ]
```

Comment:
> I'm trying to cache the workspaces but it is re-building them even though i restored from a chache.

@tylerjw, I'm not sure what config your referring to, but the logic for checking if a recovered workspace has already been successfully completed built is done here: